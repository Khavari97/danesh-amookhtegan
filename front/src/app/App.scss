@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


