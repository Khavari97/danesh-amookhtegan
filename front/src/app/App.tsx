import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
