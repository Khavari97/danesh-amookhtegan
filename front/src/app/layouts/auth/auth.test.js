import React from "react"
import Enzyme, { shallow,configure,mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from './auth';
import {Grid} from "@material-ui/core";

configure({ adapter: new Adapter() })

const setup = (props = {}) => {
  return shallow(<App {...props} />);
};

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
};

describe('render auth without error',()=>{
    let wrapper = setup();
    it('test src of images',()=>{
        wrapper.setProps({
            img: 'string',
          })
        expect(findByTestAttr(wrapper,'children').text()).toBe("<Grid />");
        expect(findByTestAttr(wrapper,'darkGreenDown').text()).toBe("");
        expect(wrapper.length).toEqual(1);
        expect(wrapper).toBeTruthy();
        expect(findByTestAttr(wrapper,'darkGreenDown').prop('src')).toBe('dark-green-down.svg');
        expect(findByTestAttr(wrapper,'darkGreenTop').prop('src')).toBe('dark-green-top.svg');
        expect(findByTestAttr(wrapper,'lightGreenDown').prop('src')).toBe('light-green-down.svg');
        expect(findByTestAttr(wrapper,'lightGreenTop').prop('src')).toBe('light-green-top.svg');
    })

    it('test Grid',()=>{
        const wrapper = shallow(<App />)
        expect(wrapper.find(Grid).exists()).toBe(false);
        expect(wrapper.find(Grid).length).toEqual(0);
    })

    it('snapshot',()=>{
        const auth = <App/>
        const snapshot = renderer.create(auth).toJSON();
        expect(snapshot).toMatchSnapshot();

    })
})