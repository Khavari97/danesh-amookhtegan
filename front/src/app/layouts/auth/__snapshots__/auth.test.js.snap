// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render auth without error snapshot 1`] = `
<section
  className="section"
>
  <div
    className="MuiGrid-root formContainer MuiGridRoot MuiGrid-container MuiGrid-spacing-xs-4"
  >
    <div
      className="MuiGrid-root muiGridRoot MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      data-test="children"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <img
        alt=""
        className="imgRight"
      />
    </div>
  </div>
  <div
    className="background"
  >
    <img
      alt="background"
      className="background1"
      data-test="darkGreenDown"
      src="dark-green-down.svg"
    />
    <img
      alt="background"
      className="background2"
      data-test="darkGreenTop"
      src="dark-green-top.svg"
    />
    <img
      alt="background"
      className="background3"
      data-test="lightGreenDown"
      src="light-green-down.svg"
    />
    <img
      alt="background"
      className="background4"
      data-test="lightGreenTop"
      src="light-green-top.svg"
    />
  </div>
</section>
`;
